### PR TITLE
More Menu Actions

### LINKED ISSUE #1123: [2.0] Add all non-moderator menu actions to posts and comments

### PR DESCRIPTION
**Pull request by Sjmarf**
Added community & user blocking, sharing and "Select Text" to posts and comments. Relies on [this MlemMiddleware PR](https://github.com/mlemgroup/MlemMiddleware/pull/25).

### CODE REVIEW

File: Mlem.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1838,8 +1858,8 @@
 			isa = XCRemoteSwiftPackageReference;
 			repositoryURL = "https://github.com/mlemgroup/MlemMiddleware.git";
 			requirement = {
-				kind = upToNextMinorVersion;
-				minimumVersion = 0.13.0;
+				branch = sjmarf/searchable;
+				kind = branch;
```
Comment (EricBAndrews) [inline Mlem.xcodeproj/project.pbxproj]: Reminder

File: Mlem/App/Models/Action/BasicAction.swift [hunk 2]
```diff
@@ -46,11 +48,24 @@ struct BasicAction: Action {
...
         self.swipeIcon1 = swipeIcon1 ?? icon
         self.swipeIcon2 = swipeIcon2 ?? icon
         self.callback = enabled ? callback : nil
     }
+    
+    func callbackWithConfirmation(navigation: NavigationLayer) {
+        if let callback {
+            if let confirmationPrompt {
+                navigation.showPopup(ActionGroup(label: "Confirm", prompt: confirmationPrompt, children: [
+                    BasicAction(id: "", isOn: false, label: "Yes", color: .red, isDestructive: true, icon: "", callback: callback)
```
Comment (EricBAndrews) [inline Mlem/App/Models/Action/BasicAction.swift]:
```suggestion
                    BasicAction(id: "", isOn: false, label: "Yes", color: Palette.main.warning, isDestructive: true, icon: "", callback: callback)
```

File: Mlem/App/Utility/Extensions/Content Models/Community1Providing+Extensions.swift [hunk 3]
```diff
@@ -64,6 +64,24 @@ extension Community1Providing {
         }
     }
     
+    func toggleBlocked(feedback: Set<FeedbackType>) {
+        if feedback.contains(.toast) {
+            if !blocked {
```
Comment (EricBAndrews) [inline Mlem/App/Utility/Extensions/Content Models/Community1Providing+Extensions.swift]: *nit* Can be more concise, also maybe like 1 cycle faster to check blocked first

```suggestion
        if !blocked, feedback.contains(.toast) {
```

File: Mlem/App/Utility/Extensions/Content Models/Person1Providing+Extensions.swift [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+//
+//  Created by Sjmarf on 02/07/2024.
+//
+
+import MlemMiddleware
+
+extension Person1Providing {
+    func toggleBlocked(feedback: Set<FeedbackType> = []) {
+        if feedback.contains(.toast) {
+            if !blocked {
```
Comment (EricBAndrews) [inline Mlem/App/Utility/Extensions/Content Models/Person1Providing+Extensions.swift]: Ditto above
```suggestion
        if !blocked, feedback.contains(.toast) {
```

**Review by EricBAndrews — CHANGES_REQUESTED** [review]
Nice!

**Comment by Sjmarf** [comment]
Closes #1123, closes #1125

**Review by EricBAndrews — APPROVED** [review]
Nice! 💯

**Pull request merged**